src/coreclr/jit/fgopt.cpp

```diff
@@ -1716,8 +1716,8 @@ bool Compiler::fgOptimizeSwitchBranches(BasicBlock* block)
             if (switchTree->gtFlags & GTF_SIDE_EFFECT)
             {
                 /* Extract the side effects from the conditional */
+                compCurBB            = block;
```

Comment:
gtExtractSideEffList uses compCurBB to set BBF_HASNULLCHECK. I hope we'll get rid of that (and compCurBB) one day.